### ISSUE TITLE
no-typos doesn't detect invalid PropTypes in static class fields

### ISSUE THREAD

**Issue by swrobel** [yannickcr/eslint-plugin-react#1677]
I've observed this with versions 7.5.1 & 7.6.1. The [docs](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-typos.md) indicate that the following should cause the following warning: `Typo in declared prop type: typo` however it doesn't.

```js
class MyComponent extends React.Component {
  static propTypes = {
    a: PropTypes.typo
  }
}
```

However, when you don't use static class fields, it produces the expected warning:

```js
class MyComponent extends React.Component {}
MyComponent.propTypes = {
  a: PropTypes.typo
};
```
